### ISSUE TITLE
Add test hook and basic tests

### ISSUE THREAD

**Issue by alanking** [irods/irods_auth_plugin_pam_interactive#6]
- [x] Test hook
- [x] Some basic tests

**Comment by alanking** [comment]
Here are some reference implementations...

https://github.com/HarryKodden/SRAM-iRODS-Showcase
https://github.com/HarryKodden/pam_otp
https://github.com/ll4strw/pam_deviceauthgrant
https://gitlab.kodden.nl/harry/PAM-OIDC-Device-Flow.git

**Comment by alanking** [comment]
For now, let's not plan on using pam_python or any other custom PAM modules. We discovered a while back that it requires python2 and it doesn't seem to be actively maintained, so we probably don't want to lean on it in our testing. Plus, we have successfully demonstrated working auth flows using standard PAM modules, so we can write tests based on these.

To start, we will emulate `pam_password` in our testing. We can move to other modules (e.g. `pam_permit`, `pam_deny`) for testing purposes by leveraging the assumed root powers of running the test hook and modifying the "irods" stack.

Further down the road, we can make the plugin's PAM stack configurable (see #37).
